rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -2515,10 +2515,22 @@ static rmw_ret_t rmw_take_int(
   const rmw_subscription_t * subscription, void * ros_message,
   bool * taken, rmw_message_info_t * message_info)
 {
-  RET_NULL(taken);
-  RET_NULL(ros_message);
-  RET_NULL(subscription);
-  RET_WRONG_IMPLID(subscription);
+  RMW_CHECK_FOR_NULL_WITH_MSG(
+    taken, "taken handle is null",
+    return RMW_RET_INVALID_ARGUMENT);
```

Comment:
@Lobotuerk hmm, `taken` is not a handle. I believe you want to use [`RMW_CHECK_ARGUMENT_FOR_NULL`](https://github.com/ros2/rmw/blob/efe3f2bf932aabd70e875228e39b843103f2c4b3/rmw/include/rmw/error_handling.h#L107). Same elsewhere.

Reply:
One of the implementations used that message with the error, I'm trying to be consistent between implementations.

Reply:
Updated for RMW_CHECK_ARGUMENT_FOR_NULL